defaults.go

```diff
@@ -96,16 +99,34 @@ func insertRecursively(into interface{}, from map[string]interface{}) {
 				// If v is null, we want to progress to the next block in case
 				// a default value exists and should be applied on top.
 				if v, ok := intoAsObject[property]; ok && v != nil {
-					switch v.(type) {
+					vr := reflect.ValueOf(v)
+					switch vr.Kind() {
 
-					case map[string]interface{}:
+					case reflect.Map:
```

Comment:
used reflect because, depending on how `intoAsObject` was defined we got either `map[string]interface{}` or `map[]`.. even though both represent the same underlying data, it was must more clear to just capture all maps. Similar situation with Slice, below